### PR TITLE
ci: Fix flaky docker rotate e2e test

### PR DESCRIPTION
**Pull request by jsturtevant**
**Reason for Change**:

The test was checking the date which wasn't in the last 20 lines because the log output is too verbose.  This checks further back in the output to ensure we always have a date stamp

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/kubernetes/kubernetes_test.go [hunk 1]
```diff
@@ -2435,6 +2435,10 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 				Skip("No windows agent was provisioned for this Cluster Definition")
 			}
 
+			if !eng.ExpandedDefinition.Properties.OrchestratorProfile.KubernetesConfig.RequiresDocker() {
+				Skip("Skip docker validations on non-docker-backed clusters")
```
Comment (marosset) [inline test/e2e/kubernetes/kubernetes_test.go:2439]: CRI log rotate went stable in v1.21
https://github.com/kubernetes/enhancements/issues/2411

Should we create an issue to investigate added a test for that with aks-engine?

Comment (marosset) [reply test/e2e/kubernetes/kubernetes_test.go:2439]: I created https://github.com/Azure/aks-engine/issues/4354

**Review by marosset — APPROVED** [review]
/lgtm

**Review by mboersma — APPROVED** [review]
/lgtm++

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**